internal/aws/awsutil/conn.go

```diff
@@ -109,90 +126,63 @@ func getProxyURL(finalProxyAddress string) (*url.URL, error) {
-		return nil, nil, awserr.New("NoAwsRegion", msg, nil)
-	}
-	s, err = cn.newAWSSession(logger, cfg.RoleARN, cfg.ExternalID, awsRegion)
-	if err != nil {
-		return nil, nil, err
+		return aws.Config{}, err
 	}
 
-	config := &aws.Config{
-		Region:                 aws.String(awsRegion),
-		DisableParamValidation: aws.Bool(true),
-		MaxRetries:             aws.Int(cfg.MaxRetries),
-		Endpoint:               aws.String(cfg.Endpoint),
-		HTTPClient:             http,
+	var options []func(*config.LoadOptions) error
+	if settings.Region != "" {
+		options = append(options, config.WithRegion(settings.Region))
+		logger.Debug("Fetch region from commandline/config file", zap.String("region", settings.Region))
 	}
-	return config, s, nil
-}
 
-// GetAWSConfig returns AWS config and session instances.
-func GetAWSConfig(logger *zap.Logger, settings *AWSSessionSettings) (awsv2.Config, error) {
-	http, err := newHTTPClient(logger, settings.NumberOfWorkers, settings.RequestTimeoutSeconds, settings.NoVerifySSL, settings.ProxyAddress)
-	if err != nil {
-		logger.Error("unable to obtain proxy URL", zap.Error(err))
-		return awsv2.Config{}, err
+	if settings.RoleARN != "" {
+		options = append(options, config.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
+			if settings.ExternalID != "" {
+				o.ExternalID = aws.String(settings.ExternalID)
+			}
+		}))
 	}
 
-	cfg, err := config.LoadDefaultConfig(context.Background())
+	cfg, err := config.LoadDefaultConfig(ctx, options...)
 	if err != nil {
-		return awsv2.Config{}, err
-	}
-
-	if settings.Region != "" {
-		cfg.Region = settings.Region
-		logger.Debug("Fetch region from commandline/config file", zap.String("region", settings.Region))
+		return aws.Config{}, err
 	}
 
 	if cfg.Region == "" {
 		logger.Error("cannot fetch region variable from config file, environment variables and ec2 metadata")
-		return awsv2.Config{}, errors.New("cannot fetch region variable from config file, environment variables and ec2 metadata")
+		return aws.Config{}, errors.New("cannot fetch region variable from config file, environment variables and ec2 metadata")
 	}
 
 	cfg.HTTPClient = http
 	cfg.RetryMaxAttempts = settings.MaxRetries
-	cfg.BaseEndpoint = aws.String(settings.Endpoint)
+	if settings.Endpoint != "" {
+		cfg.BaseEndpoint = aws.String(settings.Endpoint)
+	}
 
 	return cfg, nil
 }
 
+func CreateStaticCredentialProvider(accessKey, secretKey, sessionToken string) aws.CredentialsProvider {
+	return credentials.NewStaticCredentialsProvider(accessKey, secretKey, sessionToken)
+}
+
+func CreateAssumeRoleCredentialProvider(cfg aws.Config, roleARN, externalID string) (aws.CredentialsProvider, error) {
+	stsClient := sts.NewFromConfig(cfg)
+
+	options := func(o *stscreds.AssumeRoleOptions) {
+		if externalID != "" {
+			o.ExternalID = aws.String(externalID)
+		}
+	}
+	provider := stscreds.NewAssumeRoleProvider(stsClient, roleARN, options)
+	return aws.NewCredentialsCache(provider), nil
+}
+
```

Comment:
Why were these two functions added? Unless I'm mistaken, I don't see them used anywhere outside of unit tests